app/assets/javascripts/discourse/app/controllers/user-private-messages.js

```diff
@@ -25,19 +41,21 @@ export default class extends Controller {
   @readOnly("site.can_tag_pms") pmTaggingEnabled;
 
   get messagesDropdownValue() {
-    const parentRoute = this.router.currentRoute.parent;
-
-    if (Object.keys(parentRoute.params).length > 0) {
-      return this.router.urlFor(
-        parentRoute.name,
-        this.model.username,
-        parentRoute.params
-      );
-    } else {
-      return this.router.urlFor(parentRoute.name, this.model.username);
+    let value;
+
+    for (let i = this.messagesDropdownContent.length - 1; i >= 0; i--) {
```

Comment:
I guess you used for loop to have access to `value`. Maybe it would better to use forEach but with old way function. Something like
```
this.messagesDropdownContent.forEach(function(content) {

})
```

Reply:
I'm actually looping in reverse here which is why I can't use `forEach`.

Reply:
Thank you for clarifying :)